### PR TITLE
MQTT availability_template documentation

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add documentation for `availlability_template` a new MQTT config parameter.
For sensor and binairy_sensor platform the `encoding` parameter documentation has been updated too. Technically the `encoding` parameter applies to all MQTT platforms, but will be an undocumented feature until the `encoding` feature is enable for all value templates uses within the platform. For `sensor` and `binary_sensor` only the `state_topic` is covered and documented.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/60470
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/light.mqtt.markdown [hunk 1]
```diff
@@ -69,11 +69,19 @@ availability:
       description: An MQTT topic subscribed to receive availability (online/offline) updates.
       required: true
       type: string
+    value_template:
```
Comment (emontnemery) [inline source/_integrations/light.mqtt.markdown:72]: Please make sure json and template schemas are updated too.

File: source/_integrations/switch.mqtt.markdown [hunk 2]
```diff
@@ -33,6 +33,10 @@ availability:
   required: false
   type: list
   keys:
+    availability_template:
+      description: "Defines a [template](/docs/configuration/templating/#processing-incoming-data) to extract device's availability from the `topic`. To determine the devices's availability result of this template will be compared to `payload_available` and `payload_not_available`."
+      required: false
+      type: template
```
Comment (emontnemery) [inline source/_integrations/switch.mqtt.markdown]: This is not needed?

File: source/_integrations/vacuum.mqtt.markdown [hunk 3]
```diff
@@ -48,11 +48,19 @@ availability:
       description: An MQTT topic subscribed to receive availability (online/offline) updates.
       required: true
       type: string
+    value_template:
```
Comment (emontnemery) [inline source/_integrations/vacuum.mqtt.markdown:51]: Please make sure both schemas are updated as needed.

File: source/_integrations/light.mqtt.markdown (hunk 1 above)
Comment (jbouwh) [reply source/_integrations/light.mqtt.markdown:72]: Updated the JSON schema, cannot find any template schema here.

File: source/_integrations/switch.mqtt.markdown (hunk 2 above)
Comment (jbouwh) [reply source/_integrations/switch.mqtt.markdown]: No, it was a left over, removed it

File: source/_integrations/vacuum.mqtt.markdown (hunk 3 above)
Comment (jbouwh) [reply source/_integrations/vacuum.mqtt.markdown:51]: Done

File: source/_integrations/light.mqtt.markdown (hunk 1 above)
Comment (jbouwh) [reply source/_integrations/light.mqtt.markdown:72]: found it, fixed

**Review by emontnemery — APPROVED** [review]
Thanks, @jbouwh 👍

**Pull request merged**